src/shared/test/integration/rpc-bridge-test.js

```diff
@@ -0,0 +1,206 @@
+    });
+
+    it('should invoke Bridge method handler on every channel when calling a RPC method (with Promise)', async () => {
+      const results = await bridge.call('method1', 'params1');
+      assert.deepEqual(results.sort(), ['params1foo', 'params1bar'].sort());
+    });
+
+    it('should invoke Bridge method handler on every channel when calling a RPC method (with callback)', done => {
+      bridge.call('method1', 'params1', (err, results) => {
+        assert.deepEqual(results.sort(), ['params1foo', 'params1bar'].sort());
+        assert.isNull(err);
+        done();
+      });
+    });
+  });
+
+  context('errors and timeouts', () => {
+    it(`raises an error when the listener's callback fails`, async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(errorMessage));
+      bridge.createChannel(port1);
+      reciprocalBridge.createChannel(port2);
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'params1');
+      } catch (err) {
+        error = err;
+      }
+      assert.equal(error, errorMessage);
+    });
+
+    it('destroys the Bridge channel when a RPC message fails', async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(errorMessage));
+      bridge.createChannel(port1);
+      const channel = reciprocalBridge.createChannel(port2);
+      sandbox.stub(channel, 'destroy').callThrough();
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'params1');
+      } catch (err) {
+        error = err;
+      }
+
+      assert.called(channel.destroy);
+      assert.equal(error, errorMessage);
+      assert.deepEqual(reciprocalBridge.links, []);
+    });
+
+    it('no longer send RPC messages to a Bridge channel that has received an error', async () => {
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+      const errorMessage = 'My error';
+      bridge.on('method1', (_arg, cb) => cb(new Error(errorMessage)));
+      bridge.createChannel(port1);
+      const channel = reciprocalBridge.createChannel(port2);
+      sandbox.stub(channel, 'call').callThrough();
+
+      let error;
+      try {
+        await reciprocalBridge.call('method1', 'params1');
+      } catch (err) {
+        error = err;
+      }
+      assert.equal(error.message, errorMessage);
+
+      for (let i = 0; i < 5; ++i) {
+        const results = await reciprocalBridge.call('method1', 'params1');
+        assert.deepEqual(results, []);
+        assert.calledOnce(channel.call);
+      }
+    });
+
+    it('timeouts if the Bridge channel is not connected', done => {
+      const bridge = createBridge();
+      bridge.createChannel(port1);
+
+      bridge.call('method1', 'params1', (err, results) => {
```

Comment:
I suggest to use the Promise-based API for collecting results in all these tests. This will make them easier to update if we remove the callback-based API in future.